From b4a74c7f366e1846c26808641649cbbbbfcb8577 Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Thu, 24 Mar 2022 01:40:34 +0200
Subject: [PATCH] Workaround for compatibility with packaging >= 21

`packaging.tags` is no longer a standalone file, so the original hack
becomes problematic. As there's probably no `packaging` installed in the
virtualenv, but `pip` is guaranteed to be installed, we can use
`pip._vendor.packaging` as a quick and dirty workaround.
---
 poetry/utils/env.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/poetry/utils/env.py b/poetry/utils/env.py
index 0b91aa2239..536188d2c4 100644
--- a/poetry/utils/env.py
+++ b/poetry/utils/env.py
@@ -1410,6 +1410,10 @@ def get_supported_tags(self):  # type: () -> List[Tag]
             "from ._typing import MYPY_CHECK_RUNNING, cast",
             "MYPY_CHECK_RUNNING = False\ncast = lambda type_, value: value",
         )
+        script = script.replace(
+            "from . import _manylinux, _musllinux",
+            "from pip._vendor.packaging import _manylinux, _musllinux",
+        )
 
         script += textwrap.dedent(
             """
